### ISSUE TITLE
Paerser does not correctly handle comma-containing list elements

### ISSUE THREAD

**Issue by wiltonsr** [traefik/paerser#9]
```go
package file_test

import (
	"fmt"
	"io/ioutil"
	"log"
	"os"

	"github.com/davecgh/go-spew/spew"
	"github.com/traefik/paerser/file"
)

type ConfigExample struct {
	Foo   string
	Bar   Bar
	Great bool
}

type Bar struct {
	Sub  *Sub
	List []string
}

type Sub struct {
	Name  string
	Value int
}

func ExampleDecode() {
	tempFile, err := ioutil.TempFile("", "paeser-*.yml")
	if err != nil {
		log.Fatal(err)
	}

	defer os.RemoveAll(tempFile.Name())

	data := `
foo: aaa
bar:
  sub:
    name: bbb
    value: 6
  list:
  - "AAA"
  - "BBB,CCC"
  - "DDD"
great: true
`

	_, err = fmt.Fprint(tempFile, data)
	if err != nil {
		log.Fatal(err)
	}

	// Read configuration file

	filePath := tempFile.Name()

	config := ConfigExample{}

	err = file.Decode(filePath, &config)
	if err != nil {
		log.Fatal(err)
	}

	spew.Config = spew.ConfigState{
		Indent:                  "\t",
		DisablePointerAddresses: true,
	}
	spew.Dump(config)
}
```

## Expected Output:

```text
Output:
(file_test.ConfigExample) {
	Foo: (string) (len=3) "aaa",
	Bar: (file_test.Bar) {
		Sub: (*file_test.Sub)({
			Name: (string) (len=3) "bbb",
			Value: (int) 6
		}),
		List: ([]string) (len=2 cap=2) {
			(string) (len=3) "AAA",
			(string) (len=3) "BBB,CCC"
			(string) (len=3) "DDD",
		}
	},
	Great: (bool) true
}
```

## Got:

```text
Output:
(file_test.ConfigExample) {
	Foo: (string) (len=3) "aaa",
	Bar: (file_test.Bar) {
		Sub: (*file_test.Sub)({
			Name: (string) (len=3) "bbb",
			Value: (int) 6
		}),
		List: ([]string) (len=2 cap=2) {
			(string) (len=3) "AAA",
			(string) (len=3) "BBB"
			(string) (len=3) "CCC"
			(string) (len=3) "DDD",
		}
	},
	Great: (bool) true
}
```

**Comment by wiltonsr** [comment]
Hello @ldez 

Thank you for the adjustments.

Unfortunately, even after the changes of `paerser`, it's still removed commas in middleware.

I detailed the problem in this [issue](https://github.com/wiltonsr/ldapAuth/issues/3#issuecomment-1151355928) of my `middleware`.

I'm getting this:

```
AllowedGroups => '[ou=mathematicians dc=example dc=com║ou=italians ou=scientists dc=example dc=com]'
```

Where the expected outputs was:

```
AllowedGroups => '[ou=mathematicians,dc=example,dc=com║ou=italians,ou=scientists,dc=example dc=com]'
```

And the [`yaml`](https://github.com/wiltonsr/ldapAuth/blob/58a6e6b85a08c958ca85b5618ab25d1f93d600e5/examples/dynamic-conf/ldapAuth-conf.yml#L15) file has this list:
```yaml
AllowedGroups:
  - ou=mathematicians,dc=example,dc=com
  - ou=italians,ou=scientists,dc=example,dc=com
```

**Comment by ldez** [comment]
I think the new fix #11 will fix your problem.

**Comment by wiltonsr** [comment]
Hello @ldez,

Just confirming that `traefik v2.8.2` with `paerser v0.1.6` works as expected.

ty for the fixes.
